fastlane/lib/fastlane/actions/push_to_git_remote.rb

```diff
@@ -29,6 +29,9 @@ def self.run(params)
         # optionally add the no-verify component
         command << '--no-verify' if params[:no_verify]
 
+        # optionally add the no-verify component
+        command << '--set-upstream' if params[:set_upstream]
+
         # execute our command
         Actions.sh('pwd')
```

Comment:
Unrelated to your PR, but: 
What does this do here?

Reply:
The --set-upstream flag is needed when the branch doesn't exist in the remote, i.e. it's new. It's so the remote knows how to relate the two branches. If you try to simply push a new branch it'll complain:

    fatal: The current branch feature/test has no upstream branch.
    To push the current branch and set the remote as upstream, use

    git push --set-upstream origin feature/test

So in the suggested workflow, the branch would need to be set upstream on the remote because it's a new one.

Reply:
Haha, sorry - that comment was specifically on the `Actions.sh('pwd')` line - thus being unrelated to your PR.